pkg/controller/remotemachineset/remotemachineset_controller_test.go

```diff
@@ -259,17 +316,48 @@ func TestRemoteMachineSetReconcile(t *testing.T) {
 				},
 			})
 
-			if test.validate != nil {
-				test.validate(remoteFakeClient, t)
-			}
-
 			if err != nil && !test.expectErr {
 				t.Errorf("unexpected error: %v", err)
 			}
-
 			if err == nil && test.expectErr {
 				t.Errorf("expected error but got none")
 			}
+
+			if test.expectedRemoteMachineSets != nil {
+				rMSL, err := getRMSL(remoteFakeClient)
+				if assert.NoError(t, err) {
+					for _, eMS := range test.expectedRemoteMachineSets.Items {
+						found := false
+						for _, rMS := range rMSL.Items {
+							if eMS.Name == rMS.Name {
+								found = true
+								assert.Equal(t, *eMS.Spec.Replicas, *rMS.Spec.Replicas)
+								assert.Equal(t, eMS.Generation, rMS.Generation)
+								if !reflect.DeepEqual(eMS.ObjectMeta.Labels, rMS.ObjectMeta.Labels) {
```

Comment:
@dgoodwin I've set this up as a `MachineSetList` named `expectedRemoteMachineSets` and then compare replicas, generation, labels and annotations while also looking to see if there any unexpected/missing sets. The annotation/label comparison seems a little funky to me -- let me know if you'd like to see this simplified in any way.

Reply:
Checking labels/annotations is only going to be important if I continue to make sure we don't stomp extra additional ones. Thinking I'll add that back in.

Reply:
SGTM